xformers/components/attention/csrc/cuda/mem_eff_attention/attention_scaling_coefs_updater.h

```diff
@@ -79,15 +79,17 @@ struct RegisterOps {
       accum_t mi_row, total_row;
       BASE::iterateRows(
           lane_offset,
-          [&](int accum_m) {
-            mi_row = mi[accum_m];
-            total_row = 0.0;
-          },
+          [&](int accum_m) { mi_row = mi[accum_m]; },
           [&](int accum_m, int accum_n, int idx) {
-            frag[idx] =
-                (accum_n < max_col) ? expf(frag[idx] - mi_row) : accum_t(0.0);
-            total_row += frag[idx];
+            frag[idx] = (kFullColumns || accum_n < max_col)
+                ? expf(frag[idx] - mi_row)
+                : accum_t(0.0);
           },
+          [&](int accum_m) {});
+      BASE::iterateRows(
+          lane_offset,
+          [&](int accum_m) { total_row = 0.0; },
+          [&](int accum_m, int accum_n, int idx) { total_row += frag[idx]; },
```

Comment:
For reference: this is an additional optimization which decouples the accumulation in a separate loop